dev/NavigationView/NavigationView.cpp

```diff
@@ -154,6 +159,67 @@ NavigationView::NavigationView()
 
     Unloaded({ this, &NavigationView::OnUnloaded });
     Loaded({ this, &NavigationView::OnLoaded });
+
+    m_selectionModel.SingleSelect(true);
+    m_selectionChangedRevoker = m_selectionModel.SelectionChanged(winrt::auto_revoke, { this, &NavigationView::OnSelectionModelSelectionChanged });
+
+    m_navigationViewItemsFactory = winrt::make_self<NavigationViewItemsFactory>();
+
+    s_NavigationViewItemRevokersProperty =
+        InitializeDependencyProperty(
+            L"NavigationViewItemRevokers",
+            winrt::name_of<winrt::IInspectable>(),
+            winrt::name_of<winrt::NavigationViewItem>(),
+            true /* isAttached */,
+            nullptr /* defaultValue */);
+}
+
+void NavigationView::OnSelectionModelSelectionChanged(const winrt::SelectionModel& selectionModel, const winrt::SelectionModelSelectionChangedEventArgs& e)
+{
+    auto selectedItem = selectionModel.SelectedItem();
+
+    // Ignore this callback if:
+    // 1. the SelectedItem property of NavigationView is already set to the item
+    //    being passed in this callback. This is because the item has already been selected
+    //    via API and we are just updating the m_selectionModel state to accurately reflect the new selection.
+    // 2. Template has not been applied yet. SelectionModel's selectedIndex state will get properly updated
+    //    after the repeater finishes loading.
+    // TODO: Update SelectedItem comparison to work for the exact same item datasource scenario
+    if (m_shouldIgnoreNextSelectionChange || selectedItem == SelectedItem() || !m_appliedTemplate)
+    {
+        return;
+    }
+
+    if (IsTopNavigationView())
+    {
+        auto selectedIndex = selectionModel.SelectedIndex();
+        // If selectedIndex does not exist, means item is being deselected through API
+        auto isInOverflow = selectedIndex ? !m_topDataProvider.IsItemInPrimaryList(selectedIndex.GetAt(0)) : false;
```

Comment:
>auto isInOverflow = selectedIndex ? !m_topDataProvider.IsItemInPrimaryList(selectedIndex.GetAt(0)) : false; [](start = 8, length = 107)

Below we check that indexPath's size is greater than 0 and here we don't. I'm not sure which is correct but one of these is definitely wrong.